sympy/polys/agca/modules.py

```diff
@@ -357,7 +357,7 @@ def convert(self, elem, M=None):
             if len(tpl) != self.rank:
                 raise CoercionFailed
             return FreeModuleElement(self, tpl)
-        elif elem is 0:
+        elif isinstance(elem, int) and not isinstance(elem, bool) and elem == 0:
```

Comment:
Are all these conditions necessary? I'm not sure what this code does...

Reply:
The ``` elem ``` passed to the function can be a general object. The reason for having ``` is ``` was to compare the object type of elem and also it's literal value. 

Replacing the ``` is ``` with ``` == ``` would cause test failures on certain classes having ``` __eq__ ``` defined. 
So, I essentially had to check if the type of elem was ``` int ```. But ``` bool ``` being a subclass of ``` int ``` would also satisfy that, hence the second check.

Reply:
I wasn't happy with it either, but I couldn't come up with a cleaner way.

Reply:
I wasn't happy with it either, but I couldn't come up with a cleaner way. Unlike [this case](https://github.com/sympy/sympy/pull/16986#discussion_r291528287)
``` elem is not False and elem == 0 ``` wouldn't work, since it can be a generic object other than ```bool``` or ``` int ```.

Reply:
Another way would be using, ``` type(elem) is type(0) and elem == 0 . Should I change it to that?